src/Registry.php

```diff
@@ -136,27 +185,38 @@ public function register($type, $class, $legacy = false)
      *
      * Where possible, use {@see create()} or {@see call()} instead
      *
-     * @param string $type
-     * @return string|null
+     * @template T
+     * @param class-string<T> $type
+     * @return class-string<T>|null
      */
     public function get_class($type)
     {
-        if (!empty($this->classes[$type])) {
-            return $this->classes[$type];
+        if (array_key_exists($type, $this->legacyTypes)) {
+            // trigger_error(sprintf('"%s"(): Using argument #1 ($type) with value "%s" is deprecated since SimplePie 1.8, use class-string "%s" instead.', __METHOD__, $type, $this->legacyTypes[$type]), \E_USER_DEPRECATED);
```

Comment:
Do you want to remove or uncomment this error?

Reply:
I would uncomment the trigger_error in the last minor release.

Reply:
If I merge this PR before the release it will be part of 1.8, so in that case the error seems applicable? Up to you though.

Reply:
For consistency I would keep it as comment.

Reply:
ok I will merge this now then, I will wait for you to update the change log to include this PR before creating the release.